src/components/Accordian/Accordian.js

```diff
@@ -1,67 +1,89 @@
-import React, { useState, useEffect } from "react";
+import React, { useState } from "react";
 import PropTypes from "prop-types";
 import List from "../List";
 
-const AccordianItem = (props) => {
-    const { itemData } = props;
-    const { open, titleComponent, bodyComponent } = itemData;
-    const [isOpen, setIsOpen] = useState(open);
+const ALL = "ALL";
+
+const getDefaultOpenedItems = (items = [], defaultOpen) =>{
+    let arr = [];
+    if(items.length > 0) {
+        items.forEach((item) => {
+            const {
+                id
+            } = item;
+            if(id === defaultOpen || defaultOpen === ALL ) {
+                arr.push(id)
+            }
+        })
+    }
+    return arr;
+}
 
+const AccordianItem = (props) => {
+    const {
+        itemData,
+        allowOneOpen,
+        openedItems,
+        setOpenedItems
+    } = props;    
+    const { titleComponent, bodyComponent, id } = itemData;
+    const isOpen = openedItems.find(item => item === id) ? true: false;
     const onItemClick = () => {
-        if (typeof open === "undefined") {
-            setIsOpen(!isOpen);
+        let newList = [id];
+        if(!allowOneOpen) {
+            newList = isOpen ? openedItems.filter(item => item !== id) : [...openedItems, id]
         }
-        /** Else open and close is controlled by specific elements from the parent Accordian, based on the 'open' prop */
-    };
-
-    useEffect(() => {
-        setIsOpen(open);
-    }, [open]);
-
-    return <div className={`RCB-accordian-item ${isOpen ? "RCB-accordian-open" : "RCB-accordian-close"}`}>
-        <div className="RCB-accordian-title" onClick={onItemClick}>{React.cloneElement(titleComponent, { isOpen: isOpen })}</div>        
+        setOpenedItems(newList)
+    }
+    return <li className={`RCB-accordian-item ${isOpen ? "RCB-accordian-open" : "RCB-accordian-close"}`}>
+        <div className="RCB-accordian-title" onClick={onItemClick}>{titleComponent}</div>        
```

Comment:
pass the `isOpen` flag to the titleComponent as well just in case they want to use it for adding any logic etc.